versioning/BRM/brmtypes.cpp

```diff
@@ -447,6 +447,46 @@ void TableLockInfo::deserialize(istream& i)
     i.read((char*)&dbrootList[j], 4);
 }
 
+uint32_t TableLockInfo::getInternalSize() const
+{
+  const uint32_t nameLen = ownerName.length();
+  const uint32_t dbrootListSize = dbrootList.size();
+  return (32 + sizeof(time_t) + nameLen + (dbrootListSize * 4));
+}
+
+void TableLockInfo::serialize(std::unique_ptr<char[]>& buffer, uint32_t& offset) const
+{
+  const uint32_t nameLen = ownerName.length();
+  const uint32_t dbrootListSize = dbrootList.size();
+
+  std::memcpy(&buffer[offset], (char*)&id, 8);
```

Comment:
I am fine if you define a function `char* serializeElement(char* dst,  const char* src, size_t size)` that returns a new destination.